services/ai/nakheel/utils/token_counter.py

```
@@ -0,0 +1,22 @@
+    tiktoken = None
+
+
+def count_tokens(text: str, model: str = "gpt-4o-mini") -> int:
+    if not text:
+        return 0
+    if tiktoken is not None:
+        try:
+            encoding = tiktoken.encoding_for_model(model)
+            return len(encoding.encode(text))
+        except Exception:
+            pass
```

Ruff (0.15.5)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)